### ISSUE TITLE
Possibility of retrieving all API Keys

### ISSUE THREAD

**Issue by lucas-av7** [FusionAuth/fusionauth-issues#1570]
### Problem
Currently is only possible to retrieve a single API Key given the UID

### Solution
Add retrieve_api_keys() method and endpoints

### Alternatives/workarounds
None

### Additional context
None

### Community guidelines
All issues filed in this repository must abide by the [FusionAuth community guidelines](https://fusionauth.io/community/forum/topic/1000/code-of-conduct).

### How to vote
Please give us a thumbs up or thumbs down as a reaction to help us prioritize this feature. Feel free to comment if you have a particular need or comment on how this feature should work.

**Comment by robotdan** [comment]
This was intentional omission when we shipped the API key API. The primary reason is due to security concerns, and the large threat surface this feature would provide. 

Perhaps we can revisit and make this capability an opt-in feature and leave it off by default.

**Comment by mooreds** [comment]
@lucas-av7 can you please share more about your use case?

**Comment by lucas-av7** [comment]
@robotdan  IMO, this API for retrieving by id and retrieving all should be requested only by a key manager API key.

@mooreds In our company, we have 4 environments. The PO decided to have 4 FA apps in each environment rather than use Tenants. In order to have the same configuration in every environment, I created an automation project using python and the FA Client to act as Terraform does. I have all modules and objects data in `YAML` files.

My automation count on Object names, so I don't have the IDs in the YAML file, I get it when I retrieve all the objects. That is the reason that I need the `retrieve_api_keys()` method

The workaround was to store the API Keys Ids of each environment in Flagsmith and get them at runtime.

I just found it strange that all other modules have the option to request all objects at once.

**Comment by mooreds** [comment]
Thanks for the explanation, @lucas-av7 . As @robotdan said, this was a conscious choice due to security concerns. We'll see how the rest of the community feels about this (as shown by the number of upvotes) and think about the issues.

Thanks for your feedback!

**Comment by lucas-av7** [comment]
Thank you!!

**Comment by robotdan** [comment]
> @robotdan IMO, this API for retrieving by id and retrieving all should be requested only by a key manager API key.

I agree. Currently only a key manager API key can use the API key API. If we were open this up further to retrieve all, this same restriction would apply.

**Comment by robotdan** [comment]
Another side note, we now support pagination and search in the admin UI for API keys, but we did not ship a Search API - for the same reasons you can't retrieve all keys now. 

If we re-consider this - we'd likely just add a Search API and not add a retrieve all.

**Comment by lucas-av7** [comment]
@robotdan Cool! It would be helpful
